gconf/gconf.go

```diff
@@ -1,79 +1,62 @@
 package gconf
 
 import (
-	"encoding/json"
-	"fmt"
-	"time"
+	"reflect"
 
-	"github.com/iov-one/weave"
-	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
 )
 
 type Store interface {
 	Get([]byte) []byte
+	Set([]byte, []byte)
 }
 
-// Int returns an integer value stored under given name.
-// This function panics if configuration cannot be acquired.
-func Int(confStore Store, propName string) int {
-	var value int
-	loadInto(confStore, propName, &value)
-	return value
-}
-
-// Duration returns a duration value stored under given name.
-// This function panics if configuration cannot be acquired.
-func Duration(confStore Store, propName string) time.Duration {
-	var value time.Duration
-	loadInto(confStore, propName, &value)
-	return value
-}
-
-// String returns a string value stored under given name.
-// This function panics if configuration cannot be acquired.
-func String(confStore Store, propName string) string {
-	var value string
-	loadInto(confStore, propName, &value)
-	return value
-}
-
-// Strings returns an array of string value stored under given name.
-// This function panics if configuration cannot be acquired.
-func Strings(confStore Store, propName string) []string {
-	var value []string
-	loadInto(confStore, propName, &value)
-	return value
+func Save(db Store, src Marshaler) error {
+	key := []byte("configuration:" + pkgPath(src))
```

Comment:
I don't like this magical pkgPath.
Why not take that as a key explicitly.

Also, can we force a Validate() here as I don't see it either here or in the cash.Initializer code.

Otherwise, yes, storing like this makes sense.